pages/_app.tsx

```diff
@@ -19,17 +19,21 @@ const CustomApp = ({ Component, pageProps, categories }) => {
       gtag('config', '${config.googleAnalytics}');`}}></script>}
       {!!config.googleAnalytics && <script async src={`https://www.googletagmanager.com/gtag/js?id=${config.googleAnalytics}`}></script>}
     </Head>
-    {!pageProps.error && <Nav categories={categories} />}
+
+    {!isErrorPage && <Nav categories={categories} />}
+
     <Component {...pageProps} />
-    {!pageProps.error && <Footer />}
+
+    {!isErrorPage && <Footer />}
   </>;
 };
 
 CustomApp.getInitialProps = async context => {
   const apiRef = new API();
   const appProps = await App.getInitialProps(context);
   const categories = await apiRef.getCategories();
-  return { ...appProps, categories };
+  const isErrorPage = appProps.pageProps?.error || context.ctx.res.statusCode === 404 || false;
+  return { ...appProps, categories, isErrorPage };
 }
```

Comment:
Do you need `getInitialProps`? having it in _app will keep deoptimizing your pages that don't have SSG methods, you could move `await apiRef.getCategories();` to the pages that needed instead.

Reply:
At the moment all the pages on the blog need these categories, so instead of having code duplication across all the pages, I move the logic to the `_app`